### PR TITLE
Chore/Fix checkStyle errors

### PR DESCRIPTION
**Pull request by thilo-behnke**
### Checklist
- [x] ~I tested the changes and all related features on the test server https://artemistest.ase.in.tum.de.~
- [x] I documented my source code using the JavaDoc / JSDoc style. :rocket: 
- [x] ~I added integration test cases for the server (Spring) related to the features~
- [x] ~I added integration test cases for the client (Jest) related to the features~
- [x] ~I added screenshots/screencast of my UI changes~
- [x] ~I translated all the newly inserted strings~

### Description

:warning: Before this is merged we need to reduce the number of allowed errors for checkStyle!

How to check errors:
`./gradlew checkStyleMain`
Then open build/reports/checkstyle/main.html.

- [x] RepositoryService.java | 50 @thilo-behnke 
- [x] RepositoryResource.java | 28 @thilo-behnke 
- [x] GitService.java | 26 @thilo-behnke 
- [x] RepositoryProgrammingExerciseParticipationResource.java | 21 @thilo-behnke 
- [x] ProgrammingExerciseService.java | 19 @ungaralex 
- [x] ParticipationResource.java | 14 @ungaralex 
- [x] UserService.java | 13 @lovaiible 
- [x] BambooService.java | 11 @thilo-behnke 
- [x] FileService.java | 9 @thilo-behnke 
- [x] ParticipationService.java | 9 @maxr96 
- [x] AuthorizationCheckService.java | 7 @lovaiible 
- [x] ContinuousIntegrationService.java | 7 @lovaiible 
- [x] ConflictingResultService.java | 6 @lovaiible 
- [x] ResultService.java | 6 @maxr96 
- [x] CourseResource.java | 6 @lovaiible 
- [x] ProgrammingExerciseResource.java | 6 @lovaiible 
- [x] ResultResource.java | 6 @maxr96 
- [x] Result.java	5 @maxr96 
- [x] ProgrammingExerciseTestCaseService.java	5 @lovaiible 
- [x] VersionControlService.java	5 @lovaiible 
- [x] SerializerUtil.java	5 @ungaralex 
- [x] TextAssessmentResource.java	5 @elkbreeder 
- [x] Exercise.java	4 @maxr96 
- [x] ExerciseService.java	4 @elkbreeder 
- [x] ModelAssessmentConflictService.java	4 @ungaralex 
- [x] ClassContext.java	4 @ungaralex 
- [x] LtiService.java	4 @ungaralex 
- [x] OracleGeneratorClient.java	4 @ungaralex 
- [x] TextSubmissionResource.java	4 @elkbreeder 
- [x] ProgrammingExercise.java	3 @elkbreeder 
- [x] ExerciseRepository.java	3 @maxr96 
- [x] TokenProvider.java	3 @jpbernius 
- [x] TextBlockService.java     1 @elkbreeder 
- [x] ModelingAssessmentService.java	3 @maxr96 

### Steps for Testing

1. Log in to ArTEMiS
2. Navigate to Course Administration
3. ...

### CODE REVIEW

**Comment by thilo-behnke** [comment]
This PR has already decreased the number of doc issues from ~460 to 323.

File: src/main/java/de/tum/in/www1/artemis/web/rest/AccountResource.java [hunk 1]
```diff
@@ -101,7 +101,7 @@ public String isAuthenticated(HttpServletRequest request) {
     @GetMapping("/account")
     public UserDTO getAccount() {
         log.info("GET /account " + SecurityUtils.getCurrentUserLogin().get());
-        return Optional.ofNullable(userService.getUserWithAuthorities()).map(UserDTO::new).orElseThrow(() -> new InternalServerErrorException("User could not be found"));
+        return Optional.ofNullable(userService.getUserWithGroupsAndAuthorities()).map(UserDTO::new).orElseThrow(() -> new InternalServerErrorException("User could not be found"));
```
Comment (lovaiible) [inline src/main/java/de/tum/in/www1/artemis/web/rest/AccountResource.java]: The UserService getUserWithAuthorities() was called which was a duplicate method to getUserWithGroupsAndAuthorities(). I changed the call here and removed getUserWithAuthorities() in the UserService class.

File: src/main/java/de/tum/in/www1/artemis/web/rest/repository/TestRepositoryResource.java [hunk 2]
```diff
@@ -91,7 +91,7 @@ boolean canAccessRepository(Long exerciseId) {
      * @throws IOException
```
Comment (jpbernius) [inline src/main/java/de/tum/in/www1/artemis/web/rest/repository/TestRepositoryResource.java]: can the `@throws IOException` Comment be removed?

File: src/main/java/de/tum/in/www1/artemis/web/rest/repository/TestRepositoryResource.java [hunk 3]
```diff
@@ -104,7 +104,7 @@ boolean canAccessRepository(Long exerciseId) {
...
```
Comment (jpbernius) [inline src/main/java/de/tum/in/www1/artemis/web/rest/repository/TestRepositoryResource.java]: can the `@throws IOException` Comment be removed?

Comment (thilo-behnke) [reply src/main/java/de/tum/in/www1/artemis/web/rest/repository/TestRepositoryResource.java]: I want to do the doc updates for this file in another PR.

File: src/main/java/de/tum/in/www1/artemis/web/rest/repository/TestRepositoryResource.java (hunk 2 above)
Comment (thilo-behnke) [reply src/main/java/de/tum/in/www1/artemis/web/rest/repository/TestRepositoryResource.java]: I want to do the doc updates for this file in another PR.

**Comment by thilo-behnke** [comment]
@krusche 
This is ready to be merged.

**Pull request merged**